api/src/main/java/org/openmrs/module/ohricore/api/impl/HIVStatusComputedConcept.java

```diff
@@ -0,0 +1,72 @@
+package org.openmrs.module.ohricore.api.impl;
+
+import org.openmrs.Concept;
+import org.openmrs.Encounter;
+import org.openmrs.EncounterType;
+import org.openmrs.Obs;
+import org.openmrs.Patient;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.ohricore.api.OHRIComputedConcept;
+import org.openmrs.module.ohricore.engine.CommonsUUID;
+import org.openmrs.module.ohricore.engine.HIVStatusConceptUUID;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Date;
+import java.util.List;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+/**
+ * @author MayanjaXL, Amos, Stephen, smallGod date: 28/06/2021
+ */
+@Component("hivStatusComputedConcept")
+public class HIVStatusComputedConcept implements OHRIComputedConcept {
+	
+	@Override
+	public Obs compute(Encounter triggeringEncounter) {
+		
+		Concept hivFinalTestConcept = getConcept(HIVStatusConceptUUID.FINAL_HIV_TEST_RESULT);
+		
+		Patient patient = triggeringEncounter.getPatient();
+		List<Obs> hivTestObs = Context.getObsService().getObservationsByPersonAndConcept(patient.getPerson(),
+		    hivFinalTestConcept);
+		
+		Concept hivStatus = computeHivStatusConcept(hivTestObs);
+		return createOrUpdateObs(patient, hivStatus);
+	}
+	
+	private Concept computeHivStatusConcept(List<Obs> hivTestObs) {
+
+        Supplier<Stream<Obs>> hivTestObsStream  = hivTestObs::stream;
+        return hivTestObsStream.get()
+                .filter(obs -> obs.getValueCoded() == getConcept(CommonsUUID.POSITIVE))
+                .findAny()
+                .map(Obs::getValueCoded)
+                .orElse(hivTestObsStream.get()
+                        .filter(obs -> obs.getValueCoded() == getConcept(CommonsUUID.NEGATIVE))
+                        .filter(obs -> valueDateIsWithin90Days(obs.getValueDate()))
+                        .findAny()
+                        .map(Obs::getValueCoded)
+                        .orElse(getConcept(CommonsUUID.UNKNOWN))
+                );
+    }
+	
+	@Override
+	public EncounterType getTargetEncounterType() {
+		return null;
+	}
+	
+	@Override
+	public Concept getConcept() {
```

Comment:
Change this to getConcept(String)  so that you just pass the concept uuid to the method

Reply:
I have another getConcept function which takes a UUID. The reason we need the other without a parameter is so that each ComputedConcept implementing class implements it and provides its own ComputedConcept UUID.

    public interface OHRIComputedConcept {

        public Concept getConcept();

        public default Concept getConcept(String UUID) {
            return Context.getConceptService().getConceptByUuid(UUID);
        }
        ----
    }

Reply:
@smallgod You do not need the getConcept() method it adds no value IMO